janitor/functions/pivot.py

```diff
@@ -852,11 +855,19 @@ def _computations_pivot_longer(
 
     if len(column_names) != len(set(column_names)):
         column_names = pd.unique(column_names)
-    df = df.loc[:, column_names]
+
+    # creates a new object - essentially a copy
+    # for large dataframes, it would be more efficient
+    # to build a new dataframe from a dictionary comprehension
+    # with copy=False
+    # but that could have adverse effects downstream
+    # if the user executes a computation that mutates the
+    # original dataframe - how to act as a copy without being a copy?
+    out = df.loc[:, column_names]
```

Comment:
Sorry, bear with me: this "creates a new object" comment, is that in relation to the fact that `.loc[...]` does a copy or in relation to that fact that you renamed it `out`?

And as to your question at the end, would it eventually be resolved by the new CoW that pandas implemented since v1.5 (opt-in, but still.)

Reply:
my comment is not good enuf - I meant there is no need for an explicit copy at the start : `df = df.copy()` since on line 866, `loc` when selecting columns creates a new object - in essence a copy is made of the original dataframe. 
you are right - eventually CoW should take care of this

Reply:
any suggestions on how to rephrase the comments? @thatlittleboy

Reply:
Let me see, @samukweku!

```
# No need to copy at the beginning because `.loc[...]` will create a new object for us.
```

Regarding doing a `.copy()` at the beginning of a function - we've used it as a pattern in the past, so putting a comment here is a good idea since that's a departure from a (previously) established pattern. Small side question: should we (1) make the pattern more explicit and document departures, or (2) abandon the pattern and rely on knowledge of `pandas` API to know when copies are made?

Reply:
Knowledge of pandas here helped. If we can delegate the copy to pandas early great. If there is a chance that any of our early operations will mutate the dataframe, then we should make an explicit copy. If we are doing an operation on only the columns (like collapse_levels), then we only need a slice, since only the columns/index is mutated (and pandas creates a new index)

Reply:
There are also some scenarios where making an explicit copy isn't needed, like in the `select` functions ( since we are calling loc under d hood) or d `move` function (iloc under d hood).

Reply:
Sorry for the delayed reply, yeah I think making it clear that the copy is made by the `.loc` is good enough for me.